ELClient/ELClientWebServer.cpp

```diff
@@ -0,0 +1,407 @@
+#include "ELClientWebServer.h"
+
+typedef enum {
+  WS_LOAD=0,  // page first load
+  WS_REFRESH, // page refresh
+  WS_BUTTON,  // button press
+  WS_SUBMIT,  // form submission
+} RequestReason;
+
+typedef enum
+{
+  WEB_STRING=0,  // value type string
+  WEB_NULL,      // value type null
+  WEB_INTEGER,   // value type integer
+  WEB_BOOLEAN,   // value type boolean
+  WEB_FLOAT,     // value type float
+  WEB_JSON       // value type json
+} WebValueType;
+
+// handler descriptor
+struct _Handler
+{
+  String URL;                 // handler URL
+  WebServerCallback callback; // handler callback
+  struct _Handler * next;     // next handler
+};
+
+static ELClientWebServer * ELClientWebServer::instance = 0;
+
+ELClientWebServer::ELClientWebServer(ELClient* elc) :_elc(elc),handlers(0), arg_ptr(0) {
```

Comment:
There's one WebServer, but I let other handlers (RTC time or some), which may use the very same callback. This is why shortcutting to web-server is not good.